pkg/util/discarded_metrics.go

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
Note to reviewers: these methods used to live in the `validation` package, but they are called only from the `distributor` package (all of them) and from the `ingester` package (`DiscardedSamplesCounter()` only). Therefore I put them in a separate util source.